08-components-lab/src/03-colorpicker.jsx

```diff
@@ -11,7 +11,56 @@
  * colored <div>
  */
 
+/**
+ * Created by ofek_am Vardi LTD. on 18/12/2016.
+ */
+/**
+ * Created by ofek_am on 18/12/2016.
+ */
 import React from 'react';
 
-export default React.createClass({
-});
+export default class ColorPicker extends React.Component{
+
+    static propTypes = {
+
+    };
+
+    static defaultProps = {
+
+    };
+
+    constructor(props){
+        super(props);
+        this.onChange = this.onChange.bind(this)
+        this.onInput = this.onInput.bind(this)
+    }
+
+    state={
+        color:'green',
+        content:''
+    }
+
+    onChange(e){
+        debugger;
+        this.setState({color:e.target.value});
+    }
+
+    onInput(e){
+        this.setState({content:e.target.value});
```

Comment:
It is discouraged to change html directly. It's cool that you addressed the bonus question, but you were asked to pass text content into the colored div. You can use it just by changing content state variable and use it inside the div like the following: 

```
<div style={{background:this.state.color}} id="div1">
 { content } 
</div>
```

If you still want to embed html inside this div you use it like this:
```
<div style={{background:this.state.color}} 
       dangerouslySetInnerHTML={{ __html: content }}
/>
```
 behind the scenes when you use `dangerouslySetInnerHTML` it lets React know that the HTML inside of that component is not something it cares about.